src/output.rs

```diff
@@ -25,16 +26,21 @@ pub(crate) struct Output {
     destination: File,
     final_path: PathBuf,
     blackhole_address: IpAddr,
+    deduplicate: bool,
+    seen: HashSet<String>,
 }
 
 impl Output {
     pub fn from_config(cfg: &OutputConfig) -> anyhow::Result<Self> {
+        let seen = HashSet::new();
         match &cfg.ty {
             OutputConfigType::Hosts { include } => Ok(Self {
                 ty: OutputType::Hosts(include.to_owned()),
                 destination: tempfile()?,
                 final_path: cfg.destination.to_path_buf(),
                 blackhole_address: cfg.blackhole_address,
+                deduplicate: cfg.deduplicate,
+                seen: seen,
```

Comment:
This field, and likewise the one below it, can be initialised in-place.
```suggestion
seen: HashSet::new(),
```